### PR TITLE
Issue 5 search entries - Cody Epstein

### PR DESCRIPTION
**Pull request by kepsteen**
Added functionality for searching through entries

Recording 2024-06-05 at 15 31 23

### CODE REVIEW

File: ts/main.ts [hunk 1]
```diff
@@ -238,3 +248,27 @@ $confirmationButtons.addEventListener('click', (event: Event) => {
...
+      ) {
+        entryMatchesKeyword.push(data.entries[i]);
+      }
+    }
+    clearCardList();
+    for (let i = 0; i < entryMatchesKeyword.length; i++) {
+      $cardList.appendChild(renderEntry(entryMatchesKeyword[i]));
+    }
+  }
+});
```
Comment (brandon-moy) [inline ts/main.ts:274]: It feels like a lot more work to delete from the DOM and then recreate entries when you are clearing the filter. Why not just toggle its visibility?

**Review by brandon-moy — CHANGES_REQUESTED** [review]
Good work on the feature and how you are filtering both the title and the notes for the keyword but I feel like it take more resources to be adding and removing items than if you show and hide them.

File: ts/main.ts (hunk 1 above)
Comment (kepsteen) [reply ts/main.ts]: Recording 2024-06-06 at 09 15 01

**Review by brandon-moy — APPROVED** [review]
Great job reworking this, looks good! Approved!

**Pull request merged**